components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientsFetchResourceImpl.java

```diff
@@ -0,0 +1,222 @@
+import org.xwiki.query.QueryManager;
+import org.xwiki.rest.XWikiResource;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Default implementation for {@link PatientsFetchResource} using XWiki's support for REST resources.
+ *
+ * @version $Id$
+ * @since 1.3M5
+ */
+@Component
+@Named("org.phenotips.data.rest.internal.DefaultPatientsFetchResourceImpl")
+@Singleton
+public class DefaultPatientsFetchResourceImpl extends XWikiResource implements PatientsFetchResource
+{
+    /** Jackson object mapper to facilitate array serialization. */
+    private final ObjectMapper objectMapper = getCustomObjectMapper();
+
+    /** Logging helper object. */
+    @Inject
+    private Logger logger;
+
+    /** The query manager for patient retrieval. */
+    @Inject
+    private QueryManager qm;
+
+    /** The secure patient repository. */
+    @Inject
+    @Named("secure")
+    private PatientRepository repository;
+
+    /** XWiki request container. */
+    @Inject
+    private Container container;
+
+    @Inject
+    private Provider<Autolinker> autolinker;
+
+    @Override
+    public Response fetchPatients()
+    {
+        final Request request = this.container.getRequest();
+        // Get the internal and external IDs, if provided.
+        final List<Object> eids = request.getProperties("eid");
+        final List<Object> ids = request.getProperties("id");
+
+        this.logger.debug("Retrieving patient records with external IDs [{}] and internal IDs [{}]", eids, ids);
+
+        // Build a set of patients from the provided external and/or internal ID data.
+        final ImmutableSet.Builder<PrimaryEntity> patientsBuilder = ImmutableSet.builder();
+
+        try {
+            addEids(patientsBuilder, eids);
+            addIds(patientsBuilder, ids);
+            // Generate JSON for all retrieved patients.
+            final String json = objectMapper.writeValueAsString(patientsBuilder.build());
+            return Response.ok(json, MediaType.APPLICATION_JSON_TYPE).build();
+        } catch (final JsonProcessingException ex) {
+            logger.warn("Failed to serialize patients [{}] to JSON: {}", eids, ex.getMessage());
```

Comment:
I think this should be an `error` not a `warn` (which would be consistent with all other uses of error/warn in REST/controller code). Same in the other `catch`